### PR TITLE
fix #8684

### LINKED ISSUE #8684: Issue with "CanvasRenderingContext2D.createPattern()": need to mention that it can return null
MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/createPattern

#### What information was incorrect, unhelpful, or incomplete?

The "Return value" is not always a `CanvasPattern`.

In the special case where the source `image` is not fully loaded (`image.complete` is `false`), then `CanvasRenderingContext2D.createPattern()` returns `null`.

#### Specific section or headline?

In page "Web technology for developers > Web APIs > CanvasRenderingContext2D > CanvasRenderingContext2D.createPattern()", at section "Syntax > Return value".

#### What did you expect to see?

I expect to see both :
 - the return value for the expected case : returning a `CanvasPattern`. This is now correctly documented.
 - as well as the return value for the edge case : returning `null` when the image is not complete. This is missing from the documentation.

Knowing this could help with handling that special case.

#### Did you test this? If so, how?

I had a bug when my program used `CanvasRenderingContext2D.createPattern(img, "repeat-x")` on an image element `img` that was not always fully loaded depending on timing. When it happened, the return value was `null` and that was unexpected for me when I relied on this documentation.

I tested it in two browsers :
- Firefox 91.0.2 (64-bit) on Linux (official package for Arch Linux)
- Chromium Version 93.0.4577.63 (Official Build) Arch Linux (64-bit)

I fixed it by using the correct form described further down in the same page under section "Examples > Javascript" where an image is only used after its `load` event.

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/api/canvasrenderingcontext2d/createpattern`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/createPattern
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/canvasrenderingcontext2d/createpattern/index.html
* Last commit: https://github.com/mdn/content/commit/dfd3bb751da1f424ba5091cbcb9130293bd0b5d1
* Document last modified: 2021-09-02T01:36:33.000Z

</details>

### PR DESCRIPTION
**Pull request by smmilut**
As described in #8684, this method can return `null`.

I tried to follow guidelines but this is my very first contribution, so I hope the style fits.

I couldn't test my change since I only edited the file in GitHub.

### Description

Fix an undocumented case where the method returns `null`.

### Motivation

Now that this return value is documented, readers will be able to handle this case.

### Related issues and pull requests

Fixes #8684

### CODE REVIEW

File: files/en-us/web/api/canvasrenderingcontext2d/createpattern/index.md [hunk 1]
```diff
@@ -63,6 +63,8 @@ createPattern(image, repetition)
 
 - {{domxref("CanvasPattern")}}
   - : An opaque object describing a pattern.
+
```
Comment (hamishwillee) [inline files/en-us/web/api/canvasrenderingcontext2d/createpattern/index.md:66]: Remove whitespace github complaining about.
```suggestion

```

**Review by hamishwillee — APPROVED** [review]
Thanks for this. I removed the comment about seeing the examples below because the fail case isn't something that is covered. Otherwise looks great. You could have chosen to have `null` as a possible return value, but I prefer what you did.

I also made some other sprurious changes to layout. Please ignore :-)

**Comment by smmilut** [comment]
It works for me !

For what it's worth, the mention of the examples was because the first example provides the correct solution: only dealing with the image after it's loaded by using its `onload` callback, thus avoiding the `null` case entirely. As a beginner, this was the solution I was missing when I encountered the problem. I understand if that is outside the scope of this article, or if my first attempt was badly phrased.
If we want to include a change to this effect, maybe a simple comment in the code example could be a better way to explain this than to mention it in the "Return value". Something like:

```js
const canvas = document.getElementById("canvas");
const ctx = canvas.getContext("2d");
const img = new Image();
img.src = "canvas_createpattern.png";
img.onload = () => {  // Only use the image after it's loaded
  const pattern = ctx.createPattern(img, "repeat");
  ctx.fillStyle = pattern;
  ctx.fillRect(0, 0, 300, 300);
};
```

**Comment by hamishwillee** [comment]
> img.onload = () => {  // Only use the image after it's loaded

Yes, that could work @smmilut !
